src/components/learner-credit-management/assignments-status-chips/FailedSystem.jsx

```diff
@@ -1,29 +1,50 @@
 import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
 import { Chip, Hyperlink, useToggle } from '@edx/paragon';
 import { Error } from '@edx/paragon/icons';
 import { getConfig } from '@edx/frontend-platform/config';
 
 import BaseModalPopup from './BaseModalPopup';
+import EVENT_NAMES from '../../../eventTracking';
 
-const FailedSystem = () => {
+const FailedSystem = ({ trackEvent }) => {
   const [isOpen, open, close] = useToggle(false);
   const [target, setTarget] = useState(null);
+  const {
+    BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_FAILED_SYSTEM,
+    BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_FAILED_SYSTEM_HELP_CENTER,
+  } = EVENT_NAMES.LEARNER_CREDIT_MANAGEMENT;
+
+  const openChipModal = () => {
+    open();
+    trackEvent(BUDGET_DETAILS_ASSIGNED_DATATABLE_CHIP_FAILED_SYSTEM, { isOpen: !isOpen });
```

Comment:
Does this need to be `!!isOpen`?

Reply:
Is this one of those situations where you can't edit state and read the change in the same event loop cycle?  You called open(), but isOpen remains the old value until the next cycle?

Reply:
That is true, this particular components start is inversed at the time we read it for a segment event. 👍🏽